### PR TITLE
fix(build-tools): type test incremental build

### PR DESCRIPTION
**Pull request by jason-ha**
when validation is valid, there won't be a
prior package.json to be read for stats.
Any missing file results in non-incremental task.

### CODE REVIEW

**Review by alexvy86 — APPROVED** [review]
Good catch

File: build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/miscTasks.ts [hunk 1]
```diff
@@ -236,13 +236,19 @@ export class TypeValidationTask extends LeafWithFileStatDoneFileTask {
...
 		if (this.inputFiles === undefined) {
-			const { packageJsonPath: previousPackageJsonPath } = getTypeTestPreviousPackageDetails(
-				this.node.pkg,
-			);
-			this.inputFiles = [
-				path.join(this.node.pkg.directory, "package.json"),
-				previousPackageJsonPath,
-			];
+			this.inputFiles = [path.join(this.node.pkg.directory, "package.json")];
+			if (!(this.node.pkg.packageJson.typeValidation?.disabled === true)) {
```
Comment (tylerbutler) [inline build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/miscTasks.ts:240]: I facepalmed... I was _so close_ to getting this right the first time. Thanks for finding and fixing!

**Review by tylerbutler — APPROVED** [review]

File: build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/miscTasks.ts (hunk 1 above)
Comment (tylerbutler) [reply build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/miscTasks.ts:240]: Nit: you could just use !== true, right? Seems a little clearer to me.

Comment (tylerbutler) [reply build-tools/packages/build-tools/src/fluidBuild/tasks/leaf/miscTasks.ts:240]: Never mind... it's too early for me to code review apparently. Ignore this lunacy. :)

**Pull request merged**